aptos-move/mvhashmap/src/types.rs

```diff
@@ -169,6 +172,63 @@ impl ShiftedTxnIndex {
     }
 }
 
+// TODO[agg_v2](cleanup): consider adding `DoesntExist` variant.
+// Currently, "not existing value" is represented as Deletion.
+#[derive(Debug, PartialEq, Eq)]
+pub enum ValueWithLayout<V> {
+    // When we read from storage, but don't have access to layout, we can only store the raw value.
+    // This should never be returned to the user, before exchange is performed.
+    RawFromStorage(Arc<V>),
+    // We've used the optional layout, and applied exchange to the storage value.
+    // The type layout is Some if there is a delayed field in the resource.
+    // The type layout is None if there is no delayed field in the resource.
+    Exchanged(Arc<V>, Option<Arc<MoveTypeLayout>>),
+}
+
+impl<T> Clone for ValueWithLayout<T> {
```

Comment:
How is this different from derived clone implementation?

Reply:
Arc doesn't derive Clone for generic types without the right bound (thanks to rust expert @igor-aptos)

Reply:
hm, makes sense (so ignore the comment), but also doesn't make sense to me (from Arc implementation point of view..), why would there need to be a restriction on V to clone a pointer.. @igor-aptos do you know why? or maybe @ziaptos can know or guess, just out of curiosity.

Reply:
Not sure. I tried using derived Clone instead of this. It's giving me `the trait bound `V: std::clone::Clone` is not satisfied` errors in a bunch of places.

Reply:
yeah, very weird requirement to me, must have some reason though, and would like to know (for my own curiosity)

Reply:
from what I could find - seems the reason is:

Clone derive blindly bind T: Clone to all generic types of a struct. (even though it could be that Arc<T>: Clone no matter what T is)

In general, Clone derive (and the others in std) are meant to work only in the simple case, and try not to be too smart. but in anything more than the simple case you are expected to override it manually

Reply:
Looks like they are discussing some general features that would allow relaxing that, but who knows when that shows up.